source/configuration/sharding.txt

```diff
@@ -0,0 +1,161 @@
+preceding formats:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-formats
+   :end-before: # end-shard-key-formats
+
+Shorthand Syntax
+~~~~~~~~~~~~~~~~
+
+The shorthand syntax allows you to declare a shard key by specifying only the
+field name. This syntax supports only ranged and compound shard keys, and does not allow you
+to specify collection or sharding options.
+
+The following example creates a ranged and a compound shard key:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-shorthand
+   :end-before: # end-shard-key-shorthand
+
+Specify Associated and Embedded Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can specify a shard key on a ``belongs_to`` association in place of a field
+name. When doing so, {+odm+} creates the shard key on the primary key of the
+associated collection.
+
+The following example creates a shard key on the ``belongs_to`` association in a
+``Person`` model. Because the associated ``country`` collection has a primary
+key called ``country_id``, {+odm+} shards by that field:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-association
+   :end-before: # end-shard-key-association
+
+You can specify a shard key on an embedded document by using dot notation to
+delimit the field names. The following example creates a shard key on the
+``address.city`` field:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-embedded
+   :end-before: # end-shard-key-embedded
+
+.. note::
+
+   Because the ``.`` character is used to delimit embedded fields, {+odm+} does
+   not support creating shard keys on fields with names that contain the ``.``
+   character.
+
+Sharding Management Rake Tasks
+------------------------------
+
+You can shard collections in your database according the the shard keys defined
+in your {+odm+} models by running the ``db:mongoid:shard_collections`` Rake
+task. To ensure that the collections contain indexes that start with the shard
+key, you can first run the ``db:mongoid:create_indexes`` Rake task.
+
+Run the following Rake commands to create the indexes and shard the collections
+based on your model's shard keys:
+
+.. code-block:: bash
+
+   rake db:mongoid:create_indexes
+   rake db:mongoid:shard_collections
+
+Index management and sharding rake tasks do not stop when they encounter an
+error with a particular model class. Instead, they log the error and continue
+processing the next model. To ensure the rake tasks did not encounter any
+errors, check the output of {+odm+} logger configured for your application.
+
+.. note::
+
+   When performing schema-related operations in a sharded cluster, nodes might
+   have out of date local configuration-related cache data. To clear the caches,
```

Comment:
```suggestion
   contain out-of-date local configuration-related cache data. To clear the caches,
```

if the cache data is related to local configurations (as opposed to being local data that's related to configurations), there should be a hyphen between 'local' and 'configuration' too.

https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#hyphens-in-compound-modifiers

Reply:
I think it's local data that's configuration related